lib/nats.js

```diff
@@ -843,6 +865,9 @@ Client.prototype.publish = function(subject, msg, opt_reply, opt_callback) {
  */
 
 Client.prototype.subscribe = function(subject, opts, callback) {
+  if (this.closed) {
+    this.emit('error', SUB_CONN_CLOSED_ERR);
```

Comment:
Understood, on backward compatibility.  That's why I didn't touch `subscribe` in my PR.  I don't have a good backwardly-compatible solution.  I think it will take a separate method.